sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+
+### Setting partition key on EventData is not set in Kafka consumer
+
+The partition key of the EventHubs event is available in the kafka record headers, the protocol specific key being "x-opt-partition-key" in the header.
+
+It is by-design that we don't promote the Kafka message key to be the Event Hubs partition key and nor the reverse because with the same value the Kafka client and the Event Hub client likely send the message to two different partitions.  It might cause some confusion if we set the value in the cross protocol communication case. Exposing the properties with a protocol specific key to the other protocol client should be good enough.
+
+## Troubleshoot EventProcessorClient issues
+
+### 412 precondition failures when checkpointing
+
+412 precondition errors occur when the customer tries to perform checkpointing but the local version of the checkpoint is outdated.  This occurs when partition ownership is stolen by another instance. See [Partition ownership changes a lot](#partition-ownership-changes-a-lot) for more information.
+
+### Partition ownership changes a lot
+
+This can occur in situations where the network latency is high between storage and the client and ownership is lost to another processor instance because the `PartitionOwnershipExpirationInterval` has elapsed.  Each load balancing interval, the EventProcessorClient looks for ownership records where the expiration interval has elapsed.  If it has, it believes no one is currently processing that partition, so it will take ownership of it and start processing events from the last checkpoint.  In this scenario, it is possible to process events more than once.  One way to mitigate this is to increase `LoadBalancingUpdateInterval` and increase `PartitionOwnershipExpirationInterval`.
+
+### "...current receiver 'nil' with epoch '0' is getting disconnected"
+
+This error is expected when load balancing occurs. Load balancing is indeed an ongoing process.  When using the BlobCheckpointStore with your consumer, every ~30 seconds (by default) the consumer will check to see which consumers have a claim for each partition, then run some logic to determine whether it needs to 'steal' a partition from another consumer.  The service side mechanism used to 'steal' partitions is [Epoch][Epoch].
+
+The full error message looks something like:
+
+> New receiver 'nil' with higher epoch of '0' is created hence current receiver 'nil' with epoch '0'
+> is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used.
+> TrackingId:<GUID>, SystemTracker:<NAMESPACE>:eventhub:<EVENT_HUB_NAME>|<CONSUMER_GROUP>,
+> Timestamp:2022-01-01T12:00:00}"}
+
+### High CPU usage
+
+* Usually because too many partitions are owned.
+* Recommend no more than 3 partitions to every 1 CPU core; better to start with 1.5 partitions to each core and test increasing
```

Comment:
This feels like it trails off, was there more to this bullet point?

Reply:
Could this be combined into a single bullet about recommendations for starting number of partitions to see CPU usage before ramping up?